Simplenote/Remote.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+class Remote {
+    enum Result: Equatable {
+        static func == (lhs: Remote.Result, rhs: Remote.Result) -> Bool {
+            switch (lhs, rhs) {
+            case (.success, .success):
+                return true
+            case (.failure(let code1, _), .failure(let code2, _)):
+                return code1 == code2
+            default:
+                return false
+            }
+
+        }
+
+        case success
+        case failure(_ statusCode: Int, _ error: Error?)
+    }
```

Comment:
Can we use `https://developer.apple.com/documentation/swift/result` instead?

Since it's got Generics support, should be `Result<Int, Error>` I believe?